packages/gatsby/src/redux/actions/internal.ts

```diff
@@ -27,7 +27,7 @@ export const createPageDependency = (
     path,
     nodeId,
     connection,
-  }: { path: string; nodeId: string; connection: string },
+  }: { path: string; nodeId: string; connection?: string },
```

Comment:
Code in `gatsby/src/redux/reducers/component-data-dependencies.js` implies that `path` and `nodeId` may also be optional. Is that incorrect?

This function just forwards them and there are too many callers to `createPageDependency` for me to figure that out. But there's at least one case in this PR where this is also the case; the packages/gatsby/src/redux/actions/add-page-dependency.ts file will call `internalCreatePageDependency`, which is an import-alias for `createPageDependency` but which assumes these properties are optional as well.

I guess TS is not catching that? Or am I missing something.

We can also scrub the jsdoc like you've been doing in other places.

Reply:
Looked through the code, I don't see any way that path is optional. That reducer has it set up to handle path as an empty string, but it still has to be defined as a string type.

But you are totally right that nodeId is optional. Impressive PR catch!! Gonna call you Eagle Eyes Peter